esp-hal/src/twai/mod.rs

```diff
@@ -980,46 +988,90 @@ where
             } else {
                 0
             };
-        T::register_block()
+        self.twai
+            .register_block()
             .rx_err_cnt()
             .write(|w| unsafe { w.rx_err_cnt().bits(rec) });
 
         // Clear any interrupts by reading the status register
         cfg_if::cfg_if! {
             if #[cfg(any(esp32, esp32c3, esp32s2, esp32s3))] {
-                let _ = T::register_block().int_raw().read();
+                let _ = self.twai.register_block().int_raw().read();
             } else {
-                let _ = T::register_block().interrupt().read();
+                let _ = self.twai.register_block().interrupt().read();
             }
         }
 
         // Put the peripheral into operation mode by clearing the reset mode bit.
-        T::register_block()
+        self.twai
+            .register_block()
             .mode()
             .modify(|_, w| w.reset_mode().clear_bit());
 
         Twai {
             rx: TwaiRx {
-                _peripheral: PhantomData,
+                twai: unsafe { self.twai.clone_unchecked() },
                 phantom: PhantomData,
             },
             tx: TwaiTx {
-                _peripheral: PhantomData,
+                twai: unsafe { self.twai.clone_unchecked() },
                 phantom: PhantomData,
             },
+            twai: unsafe { self.twai.clone_unchecked() },
             phantom: PhantomData,
         }
     }
+
+    /// Convert the configuration into an async configuration.
+    fn into_async(self) -> TwaiConfiguration<'d, Async, T> {
```

Comment:
Remove this for now

Reply:
https://github.com/esp-rs/esp-hal/pull/2359/files#diff-d8403089e88b2df8eb0be3db73af0ff946613a6b5be289d7dfb7162296551597R1144